src/cmdline.rs

```diff
@@ -733,6 +737,16 @@ fn parse_install(matches: &ArgMatches) -> Result<Config> {
         .value_of("architecture")
         .expect("architecture missing");
 
+    // Uninitialized ECKD DASD's blocksize is 512, but after formatting
+    // it changes to the recommended 4096
+    // https://bugzilla.redhat.com/show_bug.cgi?id=1905159
+    #[cfg(target_arch = "s390x")]
+    {
+        if is_dasd(&device)? {
+            s390x::prepare_dasd(&device)?;
```

Comment:
I'm not very comfortable writing to the target disk from inside the command-line parser.  I recognize that formatting an unformatted disk isn't very destructive, but even so -- if we fail on a bad command-line argument, it's fair to expect that the target is unchanged.

Instead, how about we add a sector size query function in `dasd.rs` that returns `Result<Option<NonZeroU32>>`, call it here, and if it returns `Ok(Some(_))`, accept its answer instead of calling `get_sector_size()`?  That function can just call `is_formatted()` and return `4096` if false.  In other words, if we know we're going to low-level format the disk, return the sector size we plan to select when formatting.

(And by `4096` I mean a named constant.)

Reply:
I was thinking about using 4096 by default, but:
- `low-level format` takes time, for 20Gb disk about 2 minutes on my machine, so after those 2 minutes all opened network connections are closed, and `installer` exits with failure: `Caused by: Broken pipe (os error 32)`
- ECKD should be ready for use, so this case shouldn't happen on customer side . But if ECKD is `unformatted`, it's better to touch and initialize it ASAP
- FBA/ECKD could be 512 and 4096 (4096 is recommended and better for ECKD)
- FBA doesn't require low-level format

@bgilbert  , what do you think - should i modify this PR now or after FBA is merged?

Reply:
Let's go ahead and fix this bug, before FBA merges.  It seems like we only need to special-case unformatted ECKD DASDs, and everything else can use `get_sector_size()`?

And let's defer the network timeout issue for now.  AIUI osmet works for DASDs, so there's a workaround at least.  I've filed https://github.com/coreos/coreos-installer/issues/431 for that issue.

Reply:
done